detox/src/devices/EmulatorDriver.js

```diff
@@ -81,6 +82,48 @@ class EmulatorDriver extends AndroidDriver {
     await telnet.connect(port);
     await telnet.kill();
   }
+
+  async takeScreenshot(deviceId) {
+    const dst = tempfile('.png');
+    const src = '/sdcard/screenshot.png';
+    await this.adb.adbCmd(deviceId, `shell screencap ${src}`);
+    await this.adb.adbCmd(deviceId, `pull ${src} ${dst}`);
+    return dst;
+  }
+
+  async startVideo(deviceId) {
+    const adb = this.adb;
+
+    let promise = spawnRecording();
+    promise.catch(handleRecordingTermination);
+
+    this._video = promise.childProcess;
+
+    function spawnRecording() {
+      return adb.spawn(deviceId, [
+        'shell', 'screenrecord', '--verbose', '/sdcard/recording.mp4'
+      ]);
+    }
+
+    function handleRecordingTermination(result) {
+      const proc = result.childProcess;
+      if (proc.exitCode !== 0 && proc.signalCode !== 'SIGINT') {
+        promise = spawnRecording(width >>= 1, height >>= 1);
+        promise.catch(handleRecordingTermination);
+      }
+    }
+  }
+
+  async stopVideo(deviceId) {
+    if (this._video) {
+      this._video.kill(2);
+      await this.adb.adbCmd(deviceId, 'shell sleep 1');
```

Comment:
I removed this line and saw the corrupted videos you talked about.
The thing is, this slows down the tests a lot! 
I am trying to think of a way to mitigate the penalty:
1. Let's create each file on the device with a different name.
2. when test is done, create  a copy+cleanup task
3. add this task to a queue and let this queue run independently of the test lifecycle.

WDYT ?